### PR TITLE
feat: add ignoreErrors flag to LlamaParse

### PR DESCRIPTION
**Pull request by KindOfAScam**

### CODE REVIEW

**Comment by KindOfAScam** [comment]
This also fixes an error that would not create the directory if it did not exist.
#961

File: packages/llamaindex/src/readers/LlamaParseReader.ts [hunk 1]
```diff
@@ -330,49 +352,60 @@ export class LlamaParseReader extends FileReader {
...
+      for (const result of jsonResult) {
+        const jobId = result.job_id;
+        for (const page of result.pages) {
+          if (this.verbose) {
+            console.log(`> Image for page ${page.page}: ${page.images}`);
           }
+          for (const image of page.images) {
+            const imageName = image.name;
+            // Get the full path
+            let imagePath = `${downloadPath}/${jobId}-${imageName}`;
```
[the remark this answers is not in the document]
Comment (KindOfAScam) [reply packages/llamaindex/src/readers/LlamaParseReader.ts]: Done. Factored this and fetching and saving the image out

File: packages/llamaindex/src/readers/LlamaParseReader.ts [hunk 2]
```diff
@@ -278,19 +280,29 @@ export class LlamaParseReader extends FileReader {
...
   ): Promise<Document[]> {
-    // Creates a job for the file
-    const jobId = await this.createJob(fileContent, fileName);
-    if (this.verbose) {
-      console.log(`Started parsing the file under job id ${jobId}`);
-    }
+    let jobId;
+    try {
+      // Creates a job for the file
+      const jobId = await this.createJob(fileContent, fileName);
```
Comment (marcusschiesser) [inline packages/llamaindex/src/readers/LlamaParseReader.ts:286]: don't create a new jobId
```suggestion
      jobId = await this.createJob(fileContent, fileName);
```

**Comment by KindOfAScam** [comment]
@marcusschiesser Applied your review, my bad I missed that. Thanks!

**Review by marcusschiesser — APPROVED** [review]

**Pull request merged**